GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2459,21 +2459,29 @@ private void commitTemplatesConfigtoolStripMenuItem_Click(object sender, EventAr
 
         private void LoadCommitTemplates()
         {
-            CommitTemplateItem[] commitTemplates = CommitTemplateItem.LoadFromSettings();
-
             commitTemplatesToolStripMenuItem.DropDownItems.Clear();
 
+            CommitTemplateItem[] commitTemplates = CommitTemplateItem.LoadFromRegistrated();
+
+            if (commitTemplates.Length > 0)
+            {
+                foreach (CommitTemplateItem item in commitTemplates)
+                    AddTemplateCommitMessageToMenu(item, item.Name);
+                commitTemplatesToolStripMenuItem.DropDownItems.Add(new ToolStripSeparator());
+            }
+
+            commitTemplates = CommitTemplateItem.LoadFromSettings();
             if (null != commitTemplates)
             {
                 for (int i = 0; i < commitTemplates.Length; i++)
                 {
                     if (!commitTemplates[i].Name.IsNullOrEmpty())
                         AddTemplateCommitMessageToMenu(commitTemplates[i], commitTemplates[i].Name);
                 }
+                if (commitTemplates.Any(i => !i.Name.IsNullOrEmpty()))
+                    commitTemplatesToolStripMenuItem.DropDownItems.Add(new ToolStripSeparator());
```

Comment:
this is a candidate for optimisation - do the union of registered and deserialised templates
something like (pseudo-code)

```
var list = _commitTemplateManager.Registered.Select(x => new { Template = x, Name = x.Name })
                       .Union(new { Template = null, Name = "separator" })
                       .Union(CommitTemplateItem.LoadFromSettings().Select(....));
foreach (var item in commitTemplates)
{
    if (item.Template == null) {
        commitTemplatesToolStripMenuItem.DropDownItems.Add(new ToolStripSeparator());
        continue;
    }
    AddTemplateCommitMessageToMenu(item.Template, item.Name);
}
```